### PR TITLE
CHORE: Support multiple node versions

### PR DESCRIPTION
**Pull request by AdamJHall**
This PR adds in a build hook script to build multiple "flavours"  of this docker image so we can support different versions of node with the same docker files.

This will require updating our build triggers as well, we would need: 
image

I have discussed with @tvhees and we think there should also be an updated bash alias script to auto-switch between the the node 18 image and node 16 image based on the `.nvmrc` file in the project. I'll add a separate PR for that later.

### CODE REVIEW

**Comment by AdamJHall** [comment]
Also just an question to all, do we still need the `-offline` image?

**Comment by TheOrangePuff** [comment]
> Also just an question to all, do we still need the `-offline` image?

I was thinking about this the other day... I don't think we use it?

**Comment by tvhees** [comment]
> > Also just an question to all, do we still need the `-offline` image?
> 
> I was thinking about this the other day... I don't think we use it?

Not to my knowledge.

**Review by tvhees — COMMENTED** [review]
This looks like it removes version the version number from the linux image as well e.g. it will build `FROM node:16-alpine` instead of `FROM node:16.11-alpine3.14`

Just checking that's intended?

**Comment by AdamJHall** [comment]
@TheOrangePuff and @tvhees,

Ok, I've remove the offline image then as well as it simplifies the build hook and rules a bit.

**Comment by AdamJHall** [comment]
> This looks like it removes version the version number from the linux image as well e.g. it will build `FROM node:16-alpine` instead of `FROM node:16.11-alpine3.14`
> 
> Just checking that's intended?

Yes, sorry I should have mentioned that. I removed it as there is no alpine3.14 for node 18 and it doesn't look like it's necessary AFAICT, both images build fine without it.

**Review by TheOrangePuff — CHANGES_REQUESTED** [review]

File: hooks/build [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+elif [[ $DOCKER_TAG == 'offline' ]]; then
+  FLAVOUR='node18-offline'
+else
+  echo 'Unsupported tag.'
+  exit 1;
+fi
+
+case $FLAVOUR in
+  "node16")
+    NODE_TAG="16-alpine"
```
[the remark this answers is not in the document]
Comment (AdamJHall) [reply hooks/build]: Can do! Pushed up a new change to just pull it out of the tag and use it now.

**Review by TheOrangePuff — APPROVED** [review]

**Pull request merged**